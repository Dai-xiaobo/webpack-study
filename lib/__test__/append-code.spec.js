const Plugin = require('../index');


function unit(options) {
  const plugin = new Plugin(options);
  const data = {
    body: [{
      tagName: 'meta',
      voidTag: true,
      selfClosingTag: false,
      attributes: { name: 'author', content: '汉得供应链金融前端团队' },
    }, {
      tagName: 'meta',
      voidTag: true,
      selfClosingTag: false,
      attributes: { 'http-equiv': 'pragma', content: 'no-cache' },
    }],
    head: [{
      tagName: 'span',
      selfClosingTag: false,
      voidTag: false,
      innerHTML: '<a>hello word</a>',
    }],
  };
  plugin.run(data);
  expect(data).toMatchSnapshot();
}

test('添加单个body内容', () => {
  unit({
    body: {
      tagName: 'div',
      selfClosingTag: false,
      voidTag: false,
      attributes: {
        id: 'app',
      },
    },
  });
});
test('添加多个body内容', () => {
  unit({
    body: [{
      tagName: 'div',
      selfClosingTag: false,
      voidTag: false,
      attributes: {
        id: 'app',
      },
    }, {
      tagName: 'span',
      selfClosingTag: false,
      voidTag: false,
      attributes: {
        class: 'red',
      },
    }],
  });
});
test('添加单个head内容', () => {
  unit({
    head: {
      tagName: 'link',
      selfClosingTag: false,
      voidTag: true,
      attributes: { rel: 'shortcut icon', href: 'favicon.ico' },
    },
  });
});
test('添加多个head内容', () => {
  unit({
    head: [{
      tagName: 'link',
      selfClosingTag: false,
      voidTag: true,
      attributes: { rel: 'shortcut icon', href: 'favicon.ico' },
    }, {
      tagName: 'link',
      selfClosingTag: false,
      voidTag: true,
      attributes: { rel: 'shortcut icon', href: 'favicon2.ico' },
    }],
  });
});
test('添加单个before body内容', () => {
  unit({
    body: {
      before: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'favicon.ico' },
      },
    },
  });
});
test('添加单个before body和after body内容', () => {
  unit({
    body: {
      before: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before.ico' },
      },
      after: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after.ico' },
      },
    },
  });
});
test('添加多个before body和after body内容', () => {
  unit({
    body: {
      before: [{
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before1.ico' },
      }, {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before2.ico' },
      }],
      after: [{
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after1.ico' },
      }, {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after2.ico' },
      }],
    },
  });
});
test('添加单个before head内容', () => {
  unit({
    head: {
      before: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'favicon.ico' },
      },
    },
  });
});
test('添加单个before head和after head内容', () => {
  unit({
    head: {
      before: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before.ico' },
      },
      after: {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after.ico' },
      },
    },
  });
});
test('添加多个before head和after head内容', () => {
  unit({
    head: {
      before: [{
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before1.ico' },
      }, {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'before2.ico' },
      }],
      after: [{
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after1.ico' },
      }, {
        tagName: 'link',
        selfClosingTag: false,
        voidTag: true,
        attributes: { rel: 'shortcut icon', href: 'after2.ico' },
      }],
    },
  });
});
test('使用函数定制输入', () => {
  unit({
    body(tags) {
      tags.splice(1, 0, {
        tagName: 'div',
        selfClosingTag: false,
        voidTag: false,
        innerHTML: 'by fn',
      });
    },
  });
});
