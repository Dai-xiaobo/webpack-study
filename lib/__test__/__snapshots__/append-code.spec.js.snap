// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`使用函数定制输入 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "innerHTML": "by fn",
      "selfClosingTag": false,
      "tagName": "div",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加单个before body内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "href": "favicon.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加单个before body和after body内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "href": "before.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "after.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加单个before head内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "attributes": Object {
        "href": "favicon.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加单个before head和after head内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "attributes": Object {
        "href": "before.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "href": "after.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
}
`;

exports[`添加单个body内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "id": "app",
      },
      "selfClosingTag": false,
      "tagName": "div",
      "voidTag": false,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加单个head内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "href": "favicon.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
}
`;

exports[`添加多个before body和after body内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "href": "before1.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "before2.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "after1.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "after2.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加多个before head和after head内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "attributes": Object {
        "href": "before1.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "before2.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "href": "after1.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "after2.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
}
`;

exports[`添加多个body内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "id": "app",
      },
      "selfClosingTag": false,
      "tagName": "div",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "class": "red",
      },
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
  ],
}
`;

exports[`添加多个head内容 1`] = `
Object {
  "body": Array [
    Object {
      "attributes": Object {
        "content": "汉得供应链金融前端团队",
        "name": "author",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "content": "no-cache",
        "http-equiv": "pragma",
      },
      "selfClosingTag": false,
      "tagName": "meta",
      "voidTag": true,
    },
  ],
  "head": Array [
    Object {
      "innerHTML": "<a>hello word</a>",
      "selfClosingTag": false,
      "tagName": "span",
      "voidTag": false,
    },
    Object {
      "attributes": Object {
        "href": "favicon.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
    Object {
      "attributes": Object {
        "href": "favicon2.ico",
        "rel": "shortcut icon",
      },
      "selfClosingTag": false,
      "tagName": "link",
      "voidTag": true,
    },
  ],
}
`;
